pkg/queue/readiness/probe.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package readiness
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/knative/serving/pkg/queue/health"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	aggressiveProbeTimeout = 100 * time.Millisecond
+	// PollTimeout is set equal to the queue-proxy's ExecProbe timeout to take
+	// advantage of the full window
+	PollTimeout   = 10 * time.Second
+	retryInterval = 50 * time.Millisecond
+)
+
+// Probe wraps a corev1.Probe along with a logger and a count of consecutive, successful probes
+type Probe struct {
+	*corev1.Probe
+	count  int32
+	logger *zap.SugaredLogger
+}
+
+// NewProbe returns a pointer a new Probe
+func NewProbe(v1p *corev1.Probe, logger *zap.SugaredLogger) *Probe {
+	return &Probe{
+		Probe:  v1p,
+		logger: logger,
+	}
+}
+
+// IsAggressive indicates whether the Knative probe with aggressive retries should be used.
```

Comment:
This seems not be used outside the package?

Reply:
It is not used yet, but it will be in the very near future! #4600 can give you a pretty good idea of where we're going.